File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
  const set1 = new Set(nums1);
  const set2 = new Set(nums2);
  let size1 = set1.size;
  let size2 = set2.size;
  const unique1 = [];
  const unique2 = [];
  
  for (const num of set2) {
    set1.add(num);
    
    if (size1 !== set1.size) {
      unique2.push(num);
      size1++;
    }
  }
  
  for (const num of set1) {
    set2.add(num);
    
    if (size2 !== set2.size) {
      unique1.push(num);
      size2++;
    }
  }
  
  return [unique1, unique2]
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.py
class Solution:
  def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
    unique1 = set(nums1)
    unique2 = set(nums2)
    diff1 = list(unique1 - unique2)
    diff2 = list(unique2 - unique1)

    return [diff1, diff2]
